src/cli.rs

```diff
@@ -21,13 +24,25 @@ pub struct Cli {
     pub command: Option<Commands>,
 }
 
-pub fn run() {
+pub fn run() -> Result<(), Box<dyn Error>> {
     let args = Cli::parse();
     match args.command {
-        Some(Commands::Set { key, value }) => match key {
-            ConfigValue::Github => mrdog::set_config(mrdog::ConfigItem::GitHubApiToken(value)),
-            ConfigValue::Gitlab => mrdog::set_config(mrdog::ConfigItem::GitLabApiToken(value)),
-        },
+        Some(Commands::Set {
+            value,
+            name,
+            provider,
+        }) => {
+            let config_value = ConfigValue {
+                value,
+                name,
+                provider,
```

Comment:
NIT: Now we have a "global" option _provider_, which will look funny once we use the same "config" to set e.g. defualt view, etc.